modules/seqexec/server/src/main/scala/seqexec/server/Header.scala

```diff
@@ -28,14 +54,14 @@ object Header {
   def buildBoolean(get: SeqAction[Boolean], name: String ): KeywordBag => SeqAction[KeywordBag] = buildKeyword(get, name, BooleanKeyword)
   def buildString(get: SeqAction[String], name: String ): KeywordBag => SeqAction[KeywordBag]   = buildKeyword(get, name, StringKeyword)
 
-  private def bundleKeywords(inst: String, ks: Seq[KeywordBag => SeqAction[KeywordBag]]): SeqAction[KeywordBag] = {
-    val z = SeqAction(KeywordBag(StringKeyword("instrument", inst)))
+  private def bundleKeywords[A: HeaderProvider](inst: A, ks: Seq[KeywordBag => SeqAction[KeywordBag]]): SeqAction[KeywordBag] = {
+    val z = SeqAction(KeywordBag(StringKeyword("instrument", inst.name)))
```

Comment:
This is only required for DHS. I think it is better if the "instrument" keyword is added inside DhsClientHttp.setKeywords. The instrument string can be passed as a constructor parameter.
If done that way, I think the "instrument" keywords must be the first one, so it should be prepended.

Reply:
Yes, this should be isolated for DHS instruments only, or do a `SeqAction.void` for `GDSInstrument`